selfapi.js

```diff
@@ -367,6 +372,21 @@ API.prototype = {
     });
   },
 
+  // Build index routes.
+  toAPIIndex: function (basePath) {
```

Comment:
Additional nit: You might want to parse and re-build this URL to modify its actual path. Please have a look at how `testUrl` is handled in `testHandlerExample` for an example.